### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by angpysha**
This pull request introduces several key improvements to the student activities app, including expanded activity options, participant management features, enhanced frontend interactivity, improved UI styling, and comprehensive backend testing. The most important changes are grouped below by theme:

**Backend Features and API Enhancements**

* Added multiple new activities to the `activities` data structure in `src/app.py`, expanding the available options for students.
* Introduced validation to prevent duplicate signups and added a new API endpoint to unregister participants from activities, improving participant management.

**Frontend Functionality and UI Improvements**

* Updated `src/static/app.js` to display participants for each activity, including an unregister button for each participant, and ensured that the activities list refreshes after signup or unregister actions. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3L10-R51) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R96-R99) [[3]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R112-R156)
* Enhanced the UI in `src/static/styles.css` by adding hover effects for activity cards and styling for participant lists and delete buttons, making the interface more user-friendly and visually appealing. [[1]](diffhunk://#diff-090bb81494e72cfcac6b0a6065f88e9caf9d80c00b887e38f02a690dbec0f8b2R66-R71) [[2]](diffhunk://#diff-090bb81494e72cfcac6b0a6065f88e9caf9d80c00b887e38f02a690dbec0f8b2R83-R141)

**Testing and Documentation**

* Added backend test documentation to `README.md` and implemented comprehensive pytest-based tests for activity listing, signup, and unregister endpoints, including fixtures for state reset and validation of edge cases. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R15-R23) [[2]](diffhunk://#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R1-R21) [[3]](diffhunk://#diff-d9241c08affe08c89824413c5cfd8c941731defd4f3b985e597df5a6358f2122R1-R18) [[4]](diffhunk://#diff-61ec098b3ee112a73dd8fad6565ff42d7189664d993b15b6e07d6ae5c718501eR1-R59) [[5]](diffhunk://#diff-21ccd8203cab33acf5126429131ad0a82f90f5710a8dd086d8974da3fb420462R1-R61)

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -62,6 +98,25 @@ def signup_for_activity(activity_name: str, email: str):
...
+        raise HTTPException(status_code=400, detail="Student already signed up for this activity")
+
     # Add student
     activity["participants"].append(email)
     return {"message": f"Signed up {email} for {activity_name}"}
+
+
+@app.delete("/activities/{activity_name}/participants")
+def unregister_participant(activity_name: str, email: str):
+    """Unregister a student from an activity"""
```
[the remark this answers is not in the document]
Comment (angpysha) [reply src/app.py:112]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/angpysha/skills-getting-started-with-github-copilot/pull/2#discussion_r2835422976)
